go-controller/pkg/cni/helper_linux.go

```diff
@@ -309,41 +265,79 @@ func (pr *PodRequest) ConfigureInterface(namespace string, podName string, ifInf
 			klog.Warningf("Failed to clear stale OVS port %q iface-id %q: %v\n  %q", uuid, ifaceID, err, out)
 		}
 	}
-
 	ipStrs := make([]string, len(ifInfo.IPs))
 	for i, ip := range ifInfo.IPs {
 		ipStrs[i] = ip.String()
 	}
-
 	// Add the new sandbox's OVS port
 	ovsArgs := []string{
-		"add-port", "br-int", hostIface.Name, "--", "set",
-		"interface", hostIface.Name,
+		"add-port", "br-int", hostIfaceName, "--", "set",
+		"interface", hostIfaceName,
 		fmt.Sprintf("external_ids:attached_mac=%s", ifInfo.MAC),
 		fmt.Sprintf("external_ids:iface-id=%s", ifaceID),
 		fmt.Sprintf("external_ids:ip_addresses=%s", strings.Join(ipStrs, ",")),
-		fmt.Sprintf("external_ids:sandbox=%s", pr.SandboxID),
+		fmt.Sprintf("external_ids:sandbox=%s", sandboxID),
 	}
 
 	if out, err := ovsExec(ovsArgs...); err != nil {
-		return nil, fmt.Errorf("failure in plugging pod interface: %v\n  %q", err, out)
+		return fmt.Errorf("failure in plugging pod interface: %v\n  %q", err, out)
 	}
 
-	if err := clearPodBandwidth(pr.SandboxID); err != nil {
-		return nil, err
+	if err := clearPodBandwidth(sandboxID); err != nil {
+		return err
 	}
 
 	if ifInfo.Ingress > 0 || ifInfo.Egress > 0 {
-		l, err := netlink.LinkByName(hostIface.Name)
+		l, err := netlink.LinkByName(hostIfaceName)
 		if err != nil {
-			return nil, fmt.Errorf("failed to find host veth interface %s: %v", hostIface.Name, err)
+			return fmt.Errorf("failed to find host veth interface %s: %v", hostIfaceName, err)
 		}
 		err = netlink.LinkSetTxQLen(l, 1000)
 		if err != nil {
-			return nil, fmt.Errorf("failed to set host veth txqlen: %v", err)
+			return fmt.Errorf("failed to set host veth txqlen: %v", err)
+		}
+
+		if err := setPodBandwidth(sandboxID, hostIfaceName, ifInfo.Ingress, ifInfo.Egress); err != nil {
+			return err
 		}
+	}
+
+	ofPort, err := getIfaceOFPort(hostIfaceName)
+	if err != nil {
+		return err
+	}
+
+	if err = waitForPodFlows(ctx, ifInfo.MAC.String(), ifInfo.IPs, hostIfaceName, ifaceID, ofPort); err != nil {
+		return fmt.Errorf("error while waiting on flows for pod: %v", err)
+	}
+	return nil
+}
+
+// ConfigureInterface sets up the container interface
+func (pr *PodRequest) ConfigureInterface(namespace string, podName string, ifInfo *PodInterfaceInfo) ([]*current.Interface, error) {
+	netns, err := ns.GetNS(pr.Netns)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open netns %q: %v", pr.Netns, err)
+	}
+	defer netns.Close()
+
+	var hostIface, contIface *current.Interface
+
+	klog.V(5).Infof("CNI Conf %v", pr.CNIConf)
+	if pr.CNIConf.DeviceID != "" {
+		// SR-IOV Case
+		hostIface, contIface, err = setupSriovInterface(netns, pr.SandboxID, pr.IfName, ifInfo, pr.CNIConf.DeviceID)
+	} else {
+		// General case
```

Comment:
Shall we return immediately when `pr.CNIConf.DeviceID == "" && pr.IsSmartNIC == false` ? Because there is no OVS running in host mode, we know it would fail.

Reply:
Agree, better to catch this here.